### PR TITLE
Update to Chromium 96.0.4664.45

### LINKED ISSUE #1731: Updating to Chromium 96.0.4664.45
## Notes for the developer

Once you claim it, it is advisable that you create a [Draft Pull Request](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) for increased visibility:

GitHub Interface for creating Draft Pull Requests

Finally, make sure to reference this issue in your PR.

Feel free to ask questions and discuss issues here along the way. Thanks!

## Notes for others

Feel free to raise issues or questions throughout the process. However, please refrain from asking ETAs unless no visible progress has been made here or on the developer's PR for a while (e.g. 2 weeks).

### PR DESCRIPTION
**Pull request by Ahrotahn**
This PR updates to Chromium 96.0.4664.45 which has moved to stable a day early.  

Notable changes:
* The crash service uploader has been removed.  <sup>[3133431](https://chromium-review.googlesource.com/c/chromium/src/+/3133431)</sup>
* Some enterprise signals have been split off from `context_info_fetcher.cc` into `signal_utils.cc`.  <sup>[3182940](https://chromium-review.googlesource.com/c/chromium/src/+/3182940)</sup>
* Parts of `content/browser/file_system_access/file_system_access_file_writer_impl.cc` has been split into `safe_move_helper.cc`.  <sup>[3180202](https://chromium-review.googlesource.com/c/chromium/src/+/3180202)</sup>
* `components/safe_browsing/content/browser/safe_browsing_metrics_collector.cc` has been moved to `components/safe_browsing/core/`.  <sup>[3198534](https://chromium-review.googlesource.com/c/chromium/src/+/3198534)</sup>
* New safebrowsing removals due to changes made in [[3146892]](https://chromium-review.googlesource.com/c/chromium/src/+/3146892) and [[3176980]](https://chromium-review.googlesource.com/c/chromium/src/+/3176980).

Fixes #1731 

[Blog post](https://chromereleases.googleblog.com/2021/11/stable-channel-update-for-desktop.html)
[Change log](https://chromium.googlesource.com/chromium/src/+log/95.0.4638.69..96.0.4664.45?pretty=fuller&n=10000)

### CODE REVIEW

**Comment by kramred** [comment]
Thanks! Builds on macOS. 
Got [released today](https://chromereleases.googleblog.com/2021/11/stable-channel-update-for-desktop.html), one day ahead of schedule ?!

**Review by networkException — APPROVED** [review]
Builds and runs fine

**Comment by wchen342** [comment]
I have a problem
```
FAILED: obj/services/network/network_service/restricted_cookie_manager.o 
ccache ../../../../../../../../../../../usr/bin/clang++ -MMD -MF obj/services/network/network_service/restricted_cookie_manager.o.d -DIS_NETWORK_SERVICE_IMPL -DUSE_UDEV -DUSE_AURA=1 -DUSE_GLIB=1 -DUSE_NSS_CERTS=1 -DUSE_OZONE=1 -DUSE_X11=1 -DOFFICIAL_BUILD -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE -DNO_UNWIND_TABLES -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D_FORTIFY_SOURCE=2 -DNDEBUG -DNVALGRIND -DDYNAMIC_ANNOTATIONS_ENABLED=0 -DGLIB_VERSION_MAX_ALLOWED=GLIB_VERSION_2_40 -DGLIB_VERSION_MIN_REQUIRED=GLIB_VERSION_2_40 -DGOOGLE_PROTOBUF_NO_RTTI -DGOOGLE_PROTOBUF_NO_STATIC_INITIALIZER -DHAVE_PTHREAD -DU_USING_ICU_NAMESPACE=0 -DU_ENABLE_DYLOAD=0 -DUSE_CHROMIUM_ICU=1 -DU_ENABLE_TRACING=1 -DU_ENABLE_RESOURCE_TRACING=0 -DU_STATIC_IMPLEMENTATION -DICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_FILE -DWEBRTC_ENABLE_SYMBOL_EXPORT -DWEBRTC_ENABLE_OBJC_SYMBOL_EXPORT -DWEBRTC_ENABLE_AVX2 -DWEBRTC_NON_STATIC_TRACE_EVENT_HANDLERS=0 -DWEBRTC_CHROMIUM_BUILD -DWEBRTC_POSIX -DWEBRTC_LINUX -DABSL_ALLOCATOR_NOTHROW=1 -DWEBRTC_USE_BUILTIN_ISAC_FIX=0 -DWEBRTC_USE_BUILTIN_ISAC_FLOAT=1 -DWEBRTC_USE_X11 -DWEBRTC_HAVE_SCTP -DNO_MAIN_THREAD_WRAPPING -I../.. -Igen -I../../third_party/perfetto/include -Igen/third_party/perfetto/build_config -Igen/third_party/perfetto -Igen/shim_headers/zlib_shim -I../../third_party/libyuv/include -I../../third_party/jsoncpp/source/include -Igen/shim_headers/libpng_shim -Igen/shim_headers/libwebp_shim -I../../third_party/abseil-cpp -I../../third_party/boringssl/src/include -I../../third_party/protobuf/src -Igen/protoc_out -Igen/net/third_party/quiche/src -I../../net/third_party/quiche/overrides -I../../net/third_party/quiche/src/common/platform/default -I../../net/third_party/quiche/src -I../../third_party/ced/src -I../../third_party/icu/source/common -I../../third_party/icu/source/i18n -I../../third_party/webrtc_overrides -I../../third_party/webrtc -Igen/third_party/webrtc -fno-delete-null-pointer-checks -fno-ident -fno-strict-aliasing --param=ssp-buffer-size=4 -fstack-protector -fno-unwind-tables -fno-asynchronous-unwind-tables -fPIC -pthread -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../../tools/clang/crashreports -mllvm -instcombine-lower-dbg-declare=0 -flto=thin -fsplit-lto-unit -fwhole-program-vtables -m64 -march=x86-64 -msse3 -ffile-compilation-dir=. -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code-aggressive -Wthread-safety -Wextra-semi -Wno-missing-field-initializers -Wno-unused-parameter -Wloop-analysis -Wno-unneeded-internal-declaration -Wenum-compare-conditional -Wno-psabi -Wno-ignored-pragma-optimize -Wno-builtin-assume-aligned-alignment -Wno-unused-but-set-parameter -Wno-unused-but-set-variable -Wno-bitwise-instead-of-logical -O2 -fdata-sections -ffunction-sections -fno-unique-section-names -fno-omit-frame-pointer -g0 -ftrivial-auto-var-init=pattern -fsanitize=cfi-vcall -fsanitize-blacklist=../../tools/cfi/ignores.txt -fsanitize=cfi-icall -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wexit-time-destructors -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/sysprof-4 -DPROTOBUF_ALLOW_DEPRECATED=1 -I/usr/include/nss3 -I/usr/include/nspr4 -std=c++14 -fno-trigraphs -Wno-trigraphs -fno-exceptions -fno-rtti -fvisibility-inlines-hidden -Wno-deprecated-declarations -c ../../services/network/restricted_cookie_manager.cc -o obj/services/network/network_service/restricted_cookie_manager.o
warning: unknown warning option '-Wno-bitwise-instead-of-logical' [-Wunknown-warning-option]
../../services/network/restricted_cookie_manager.cc:164:30: error: no viable overloaded operator[] for type 'network::CookieAccessesByURLAndSite' (aka 'map<pair<GURL, net::SiteForCookies>, unique_ptr<set<net::CookieWithAccessResult, network::CookieWithAccessResultComparer>>>')
      recent_cookie_accesses_[std::tie(url, site_for_cookies)];
      ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../../../../../../../../../../../usr/bin/../lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/stl_map.h:492:7: note: candidate function not viable: no known conversion from 'tuple<const GURL &, const net::SiteForCookies &>' to 'const std::map<std::pair<GURL, net::SiteForCookies>, std::unique_ptr<std::set<net::CookieWithAccessResult, network::CookieWithAccessResultComparer>>>::key_type' (aka 'const std::pair<GURL, net::SiteForCookies>') for 1st argument
      operator[](const key_type& __k)
      ^
../../../../../../../../../../../usr/bin/../lib/gcc/x86_64-redhat-linux/11/../../../../include/c++/11/bits/stl_map.h:512:7: note: candidate function not viable: no known conversion from 'tuple<const GURL &, const net::SiteForCookies &>' to 'std::map<std::pair<GURL, net::SiteForCookies>, std::unique_ptr<std::set<net::CookieWithAccessResult, network::CookieWithAccessResultComparer>>>::key_type' (aka 'std::pair<GURL, net::SiteForCookies>') for 1st argument
      operator[](key_type&& __k)
      ^
1 warning and 1 error generated.
```

I tried
```
-      recent_cookie_accesses_[std::tie(url, site_for_cookies)];
+      recent_cookie_accesses_[static_cast<std::pair<GURL&, net::SiteForCookies&>>(std::tie(url, site_for_cookies))];
```
but that didn't work. Any idea on how to fix this?

**Comment by DAINRA** [comment]
@wchen342 try [this patch](https://github.com/void-linux/void-packages/blob/master/srcpkgs/chromium/patches/chromium-96-RestrictedCookieManager-tuple.patch)

**Comment by wchen342** [comment]
@DAINRA It works. Thanks!

**Comment by PF4Public** [comment]
Builds and runs well for me as well

**Comment by PF4Public** [comment]
Did anyone notice crazyness with PDFs?
image

**Comment by networkException** [comment]
Looks normal on my build: 
image

**Comment by PF4Public** [comment]
@networkException Do you have higher than 90 dpi? I noticed this glitch on a 123-dpi monitor.

**Comment by networkException** [comment]
`font-family` is simply set to `sans-serif` for the first next node I inspected (shame on chromium that it doesn't support devtools in pdfs)

**Comment by PF4Public** [comment]
Interesting is that newer PDFs exhibit no issues (laptop screen 142-dpi):
image

Nah, whatever, not a big issue anyway.

I thought @Zoraver will review this PR. What you think, would it be OK to merge with his thumbs-up? :)

**Comment by networkException** [comment]
Sure

**Review by PF4Public — APPROVED** [review]
Reading diff of a diff is something special :)

**Comment by PF4Public** [comment]
@networkException We are good to merge, roll on!

**Comment by networkException** [comment]
Oh shoot I missed the `-1` in the tag name, sorry for that

**Comment by Zoraver** [comment]
> I thought @Zoraver will review this PR. What you think, would it be OK to merge with his thumbs-up? :)

I am in the process of reviewing this pull request; I have had a busy week...

So far, everything looks good; if I run into anything that I think could be improved, it can be addressed in a future release.

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch [hunk 1]
```diff
@@ -381,6 +463,81 @@
  
    std::vector<std::string> GetOnSecurityEventProviders();
  
+--- a/chrome/browser/enterprise/signals/signals_utils.cc
++++ b/chrome/browser/enterprise/signals/signals_utils.cc
+@@ -10,7 +10,6 @@
+ #include "components/policy/content/policy_blocklist_service.h"
+ #include "components/policy/core/browser/url_blocklist_manager.h"
+ #include "components/prefs/pref_service.h"
+-#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
```
Comment (Zoraver) [inline patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch:472]: The removal of `components/safe_browsing/core/common/safe_browsing_prefs.h` should be dealt with in `remove-unused-preferences-fields.patch`.

File: patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch [hunk 2]
```diff
@@ -381,6 +463,81 @@
...
+ absl::optional<bool> GetChromeCleanupEnabled(PrefService* local_state) {
+   DCHECK(local_state);
+ #if defined(OS_WIN)
+--- a/chrome/browser/enterprise/signals/signals_utils.h
++++ b/chrome/browser/enterprise/signals/signals_utils.h
+@@ -5,8 +5,6 @@
+ #ifndef CHROME_BROWSER_ENTERPRISE_SIGNALS_SIGNALS_UTILS_H_
+ #define CHROME_BROWSER_ENTERPRISE_SIGNALS_SIGNALS_UTILS_H_
+ 
+-#include "components/safe_browsing/core/common/safe_browsing_prefs.h"
```
Comment (Zoraver) [inline patches/core/ungoogled-chromium/fix-building-without-safebrowsing.patch:523]: See comment above.

**Review by Zoraver — APPROVED** [review]
On the whole, this looks good to me. The issues I commented on are minor and could be addressed in a future pull request.

**Pull request merged**